core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -109,6 +116,13 @@ private[spark] class TaskSetManager(
   private val executorDecommissionKillInterval =
     conf.get(EXECUTOR_DECOMMISSION_KILL_INTERVAL).map(TimeUnit.SECONDS.toMillis)
 
+  private[scheduler] val inefficientTaskCalculator =
+    if (conf.get(SPECULATION_ENABLED) && conf.get(SPECULATION_EFFICIENCY_ENABLE)) {
```

Comment:
With https://github.com/apache/spark/pull/36162/files#r885262650, I think we can use `efficientTaskCalcualtionEnabled` directly.